### PR TITLE
pivot pvlive/all format

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Add option for reduced format for pvlive
Returned object goes from 2MB to ~400kB
Good to first merge #274 

50% of #268 

## How Has This Been Tested?

- ran locally
- add test

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/database.py [hunk 1]
```diff
@@ -32,7 +32,12 @@
 from nowcasting_datamodel.save.update import N_GSP
 from sqlalchemy.orm.session import Session
 
-from pydantic_models import GSPYield, LocationWithGSPYields
+from pydantic_models import (
+    GSPGenerations,
```
Comment (braddf) [inline src/database.py:36]: Could we maybe name this something more specific to what the return type is/looks like? Something like `GSPYieldsGroupByDatetime` or just `GSPYieldsByDatetime`

File: src/pydantic_models.py [hunk 2]
```diff
@@ -43,3 +43,53 @@ def from_location_sql(self):
...
+) -> List[GSPGenerations]:
+    """Change LocationSQL to list of OneDatetimeGSPGeneration
+
+    This converts a list of location objects to a list of OneDatetimeGSPGeneration objects.
+
+    N locations, which T gsp yields each,
+    is converted into
+    T OneDatetimeGSPGeneration objects with N gsp yields each.
+
+    This reudces the size of the object as the datetimes are not repeated for each gsp yield.
```
Comment (braddf) [inline src/pydantic_models.py:70]: small typo on `reduces`

File: src/database.py (hunk 1 above)
Comment (peterdudfield) [reply src/database.py:36]: yea good idea

File: src/pydantic_models.py (hunk 2 above)
Comment (peterdudfield) [reply src/pydantic_models.py:70]: thanks

**Comment by peterdudfield** [comment]
Hope its ok to merge, please add comments if there are any - thanks @braddf for the review

**Pull request merged**